src/server/pfs/server/driver.go

```diff
@@ -1446,14 +1446,18 @@ reloadCommitSet:
 	}
 }
 
-func (d *driver) listCommitSet(ctx context.Context, cb func(*pfs.CommitSetInfo) error) error {
+func (d *driver) listCommitSet(ctx context.Context, project *pfs.Project, cb func(*pfs.CommitSetInfo) error) error {
+	projectName := project.GetName()
 	// Track the commitsets we've already processed
 	seen := map[string]struct{}{}
 
 	// Return commitsets by the newest commit in each set (which can be at a different
 	// timestamp due to triggers or deferred processing)
 	commitInfo := &pfs.CommitInfo{}
 	err := d.commits.ReadOnly(ctx).List(commitInfo, col.DefaultOptions(), func(string) error {
+		if commitInfo.GetCommit().GetBranch().GetRepo().GetProject().GetName() != projectName {
```

Comment:
since the project name is only used once, can we just inline it?
```suggestion
		if commitInfo.GetCommit().GetBranch().GetRepo().GetProject().GetName() != project.GetName() {
```